### ISSUE TITLE
Can rotate encryption keys for RKE2 clusters

### ISSUE THREAD

**Issue by snasovich** [rancher/dashboard#4540]
**Detailed Description**

Following the implementation of https://github.com/rancher/rancher/issues/35436, add UI similar to the following that exists for RKE1 clusters to rotate encryption keys.

Note: Secrets encryption is enabled for RKE2 clusters by default and there is no option to turn if off.

![image](https://user-images.githubusercontent.com/85187633/141045814-bf0fde42-892c-44b6-9430-0d9ff340b7fc.png)

**Context**
This is needed for RKE2 provisioning parity with RKE1 provisioning.

**Comment by vincent99** [comment]
Last I heard encryption was intentionally always on for rke2 (possibly despite a flag that sounded like you could disable it but would error if you tried to use it).

**Comment by snasovich** [comment]
@vincent99 , you're correct. I'm updating the description to leave only keys rotation option.

**Comment by catherineluse** [comment]
I believe no changes are needed in the UI for this issue because the feature will be enabled if `rotateEncryptionKey` is included in the action links for the cluster data.

The Rancher API at `<rancher-server-url>/v3/clusters/<cluster-id>` currently shows these available actions for RKE2 clusters:

```
"actions":{
      "apply":"https://192.168.0.9:8005/v1/management.cattle.io.clusters/c-m-b4qzzdqg?action=apply",
      "enableMonitoring":"https://192.168.0.9:8005/v3/clusters/c-m-b4qzzdqg?action=enableMonitoring",
      "generateKubeconfig":"https://192.168.0.9:8005/v3/clusters/c-m-b4qzzdqg?action=generateKubeconfig",
      "importYaml":"https://192.168.0.9:8005/v3/clusters/c-m-b4qzzdqg?action=importYaml"
   }
```
Whereas for RKE1 clusters that are provisioned with the advanced option **Secrets Encryption** enabled, the following action is also included:

```
"rotateEncryptionKey": "…/v3/clusters/c-47qmk?action=rotateEncryptionKey",
```
That same action needs to be included in the actions for RKE2 clusters.

**Comment by catherineluse** [comment]
I was wrong - Steve doesn't support actions so this will need to be handled by updating the cluster resource in Steve, per this comment https://github.com/rancher/dashboard/issues/4485#issuecomment-1032932115

**Comment by gaktive** [comment]
@catherineluse from what I can tell from the linked comment, https://github.com/rancher/dashboard/issues/4540 has been merged so was there something else needed on the backend to unblock us here? If so, is there a ticket or request somewhere to do this already or do we need a new one?

**Comment by catherineluse** [comment]
We are currently waiting for fixes for these failing tests. https://github.com/rancher/rancher/issues/35436#issuecomment-1054657295

The fixes shouldn't require changing the PR on the UI side because the API won't change. I just put it in the backend blocked column because I thought the UI would throw an error until that was fixed, so it can't be completely tested yet.

**Comment by catherineluse** [comment]
I think before this will work, we will also have to fix a bug where snapshots aren't being loaded for RKE2 clusters, even though they have been successfully created. The rotate encryption keys modal won't let you rotate keys if you haven't created a snapshot.

- https://github.com/rancher/dashboard/issues/5131
- https://github.com/rancher/dashboard/issues/4606#issuecomment-1063475954

**Comment by snasovich** [comment]
@gaktive @catherineluse , there is some miscommunication here. https://github.com/rancher/dashboard/issues/5379 was supposed to be the ticket to ensure "Rotate Encryption Keys" button is NOT available (hidden) - I will update the description.
Now, I see that the linked PR was not merged yet, so I guess it means the button was never actually introduced and QA only did backend testing.
If that's the case, please close https://github.com/rancher/dashboard/issues/5379 as not needed and this one should be moved to 2.6.5 (will do that now).
cc: @Sahota1225

**Comment by snasovich** [comment]
@catherineluse @gaktive @nwmac , we're still putting finishing touches over backend support, but it's all "behind the scenes" types of changes and the contract UI uses for kicking off encryption key rotation won't change.
The actual functionality is already in 2.6.4 as far as backend goes (even though it's not working in all cases), so you should be able to continue with your changes.

**Comment by gaktive** [comment]
QA waiting on this ticket now for testing. May need someone other than @catherineluse to work on this in light of other urgent 2.6.5 work.

**Comment by aalves08** [comment]
@gaktive took a look into this today and we still don't available the action **rotateEncryptionKey** that @catherineluse  mentioned in her comment in Feb 8 for RKE2 clusters.

@snasovich can you confirm this? Or am I getting this completely wrong? (sorry for that if that's the case)

On the FE side, the enable of the action is dependent on that factor. (we also have it "locked" to RKE1 clusters because until now it hasn't been available for RKE2, but that's a one line change).

FYI: @richard-cox

**Comment by catherineluse** [comment]
> we still don't available the action rotateEncryptionKey

Yes, that's correct. Actions are only available on the back end through the Norman API, whereas this key rotation support is only in Steve.

Instead of using an action, the key rotation is triggered by updating the cluster YAML, which is what my PR does here: https://github.com/rancher/dashboard/pull/5144/files#diff-77e0621fe29001893f1d9591e17d8a9bbcbf73f8ebde56bec6bb3b46db88dc22R104

**Comment by catherineluse** [comment]
The current status of my PR is that Giuseppe reviewed it and he said it worked. But he requested a change to make it disabled while the action is in process. I will work on it next week

**Comment by snasovich** [comment]
@catherineluse , we're delaying this feature on back-end per https://github.com/rancher/rancher/issues/35436#issuecomment-1109111044. Could we make sure that it won't be available in UI in 2.6.5? If the majority of the code is already complete it may be a good idea to keep it in, but have it turned off - provided there is a minimal chance of regressions from it.

cc: @gaktive @nwmac

**Comment by catherineluse** [comment]
Since my PR was already merged, I created another PR that turns the feature off. It just sets the action to false instead of checking if it should be available. https://github.com/rancher/dashboard/pull/5709

**Comment by richard-cox** [comment]
@sowmyav27 The UI side of this feature was marked as a release blocker, but the dependent backend was not and now won't be delivered in 2.6.5 (see https://github.com/rancher/rancher/issues/35436#issuecomment-1109111044). Can the label be removed?

**Comment by catherineluse** [comment]
and should the milestone of this issue be changed to 2.6.6?

**Comment by gaktive** [comment]
Pushing this work to 2.6.6. Will file a corresponding 2.6.5 ticket for the thing we had to do for QA tracking.

**Comment by timhaneunsoo** [comment]
### Test Environment: ###
Rancher version: v2.6-head
Rancher cluster type: HA
Docker version: 20.10

Downstream cluster type: RKE2

----------

### Testing: ###

Tested this issue with the following steps:
1. Provision RKE2 node driver cluster
2. Go to `Cluster Management` and click on 3-dot options button on the right side of the cluster
3. Click on `Rotate Encryption Keys`

**Result** - Pass
Now user is given option to rotate encryption keys
![image.png](https://zube.io/files/rancher/33d5d75d178bd3982bb27da271f60e11-image.png)
